FastTimeSeries/src/models/RNN.jl

```diff
@@ -0,0 +1,39 @@
+function tabular2rnn(X::AbstractArray{Float32, 3})
+    X = permutedims(X, (1, 3, 2))
+    return X
+end
+
+struct RNNModel{A, B}
+    recbackbone::A
+    finalclassifier::B
+end
+
+"""
+    RNNModel(recbackbonem, outsize, recout[; kwargs...])
+
+Creates a RNN model from the recurrent 'recbackbone' architecture. The output from this backbone
+is passed through a dropout layer before a 'finalclassifier' block.
+
+## Keyword arguments.
+
+- `outsize`: The output size of the final classifier block. For single classification tasks,
+    this would be the number of classes.
+- `recout`: The output size of the `recbackbone` architecture.
+- `dropout_rate`: Dropout probability for the dropout layer.
+"""
+
+function RNNModel(recbackbone;
+                outsize,
+                recout,
+                kwargs...)
+    return RNNModel(recbackbone, Dense(recout, outsize))
+end
+
+function (m::RNNModel)(X)
+    X = tabular2rnn(X)
+    Flux.reset!(m.recbackbone)
```

Comment:
is it intentional to have the initial state trainable? If not, I would add `ChainRulesCore` as a dependency and wrap the `reset!` call like so:
```suggestion
    ChainRulesCore.ignore_derivatives() do
        Flux.reset!(m.recbackbone)
    end
```